src/main/java/org/sonar/plugins/checkstyle/metadata/CheckUtil.java

```diff
@@ -34,6 +34,8 @@
 import com.puppycrawl.tools.checkstyle.utils.TokenUtil;
 
 public final class CheckUtil {
+    private static PackageObjectFactory packageObjectFactory;
```

Comment:
Util class should stay stateless.
Let's move this factory to class that calls getCheck

Reply:
Done.